exopy_qm/tasks/tasks/ConfigureExecuteTask.py

```diff
@@ -0,0 +1,264 @@
+
+    - get_config(parameters)/get_program(parameters) for the
+    configuration file and the program file respectively. The
+    parameters argument is a dictionnary containing the values entered
+    by the users and sould be converted to the appropriate python type
+    before using it.
+
+    The two files can be merged into one if wanted.
+
+    """
+
+    #: Path to the python configuration file
+    path_to_config_file = Unicode().tag(pref=True)
+
+    #: Module containing the configuration file
+    _config_module = Value()
+
+    #: Module containing the program file
+    _program_module = Value()
+
+    #: Path to the python program file
+    path_to_program_file = Unicode().tag(pref=True)
+
+    #: Maximum duration allowed for the QM
+    duration_limit = Long(default=int(500000)).tag(pref=True)
+
+    #: Maximum amount data allowed for the QM
+    data_limit = Long(default=int(7000000)).tag(pref=True)
+
+    #: Waiting time before fetching the results from the server
+    wait_time = Float(default=1.0).tag(pref=True)
+
+    #: Parameters entered by the user for the program and config
+    parameters = Typed(dict).tag(pref=True)
+
+    #: Comments associated with the parameters
+    comments = Typed(dict).tag(pref=True)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._config_module = None
+        self._program_module = None
+        self.parameters = {}
+        self.comments = {}
+
+    def check(self, *args, **kwargs):
+        test, traceback = super(ConfigureExecuteTask,
+                                self).check(*args, **kwargs)
+
+        if not test:
+            return test, traceback
+
+        if self._config_module is None or self._program_module is None:
+            msg = ('Config or program missing')
+            traceback[self.get_error_path() + '-trace'] = msg
+
+        for key, value in self.parameters.items():
+            try:
+                self.format_and_eval_string(value)
+            except:
+                e = sys.exc_info()[0]
+                msg = ("Couldn't evaluate {} : {}")
+                traceback[self.get_error_path() + '-trace'] = msg.format(value, e)
+
+        return test, traceback
+
+    def perform(self):
+        self._update_parameters()
+
+        # Evaluate all parameters
+        evaluated_parameters = {}
+        for key, value in self.parameters.items():
+            evaluated_parameters[key] = self.format_and_eval_string(value)
+
+        config_to_set = self._config_module.get_config(evaluated_parameters)
+        program_to_execute = self._program_module.get_prog(
+            evaluated_parameters)
+
+        self.driver.set_config(config_to_set)
+        self.driver.execute_program(
+            program_to_execute, self.duration_limit, self.data_limit)
+        time.sleep(self.wait_time)
+        results = self.driver.get_results()
+
+        for k in results.variable_results.__dict__:
```

Comment:
Is acceding the instance dict truly the only to do this ? This seems weird.

Reply:
I took the code from the GetResultsTask module. I don't think there is any other way to do this since the name of the members are dynamically generated.